### DISCUSSION TITLE
shouldFocusError with multiline TextInput not working?

### DISCUSSION THREAD

**Discussion by msageryd** [react-hook-form/react-hook-form#7604] [Q&A] [answered]
I can't get `shouldFocusError` to work with TextInput where `multiline=true`.

I suspect that the `ref` need to be set to something else with multiline inputs? When using multiline there seems to be a scrollView inside of the TextInput. I suspect that this needs to be taken into account when assigning `ref`, but I have no idea how to do this.

Is focusong multiline inputs a known problem (I didn't find any discussion or issue on this), or might it be a React-Native problem?

Any suggestion on what to try?

**Comment by Moshyfawn** [comment, marked as answer]
Can't reproduce. See [this snack](https://snack.expo.dev/@moshyfawn/rn-multiline-error-field-focus)

**Reply by msageryd** [reply]
Man, you are fast =)
Yeah, you are correct. It works as it should. I just discovered that my problem is elsewhere. The field is focused correctly, but my scrollView does not pick up the new focus so it can scroll to it. This resulted in the focus (and the cursor) sitting behind the keyboard. 

Either something is broken with RNs auto-scroll to foused field when multiline is involved, or my keybboard-handling is broken. It's not always a joy to be a programmer..

**Comment by msageryd** [comment]
For anyone ending up here searching for the same problem:
This seems to be a ScrollView problem with multiline TextInput. I suspect that the inner scrollview used for multiline is giving some trouble.

Disabling the inner scrollview solves the problem. I only have at most a couple of lines of text, so it's fine without the inner scroll capability.

```
<TextInput
   scrollEnabled={false}
>
```

edit: 
I just found this lib
https://github.com/baijunjie/react-native-input-scroll-view

If I will have need for scrolling inside of a multiline TextInput I'll look closer into this. Either use the lib or look how they solved the problem.
